src/OpStateBridge.sol

```diff
@@ -0,0 +1,205 @@
+contract OpStateBridge is Ownable2Step {
+    ///////////////////////////////////////////////////////////////////
+    ///                           STORAGE                           ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice The address of the OpWorldID contract on any OP Stack chain
+    address public immutable opWorldIDAddress;
+
+    /// @notice address for OP Stack chain Ethereum mainnet L1CrossDomainMessenger contract
+    address internal immutable crossDomainMessengerAddress;
+
+    /// @notice Ethereum mainnet worldID Address
+    address public immutable worldIDAddress;
+
+    /// @notice Amount of gas purchased on the OP Stack chain for _propagateRoottimism
+    uint32 internal _gasLimitSendRootOp;
+
+    /// @notice Amount of gas purchased on the OP Stack chain for setRootHistoryExpiryOptimism
+    uint32 internal _gasLimitSetRootHistoryExpiryOp;
+
+    /// @notice Amount of gas purchased on the OP Stack chain for transferOwnershipOptimism
+    uint32 internal _gasLimitTransferOwnershipOp;
+
+    ///////////////////////////////////////////////////////////////////
+    ///                            EVENTS                           ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice Emmitted when the the StateBridge gives ownership of the OPWorldID contract
+    /// to the WorldID Identity Manager contract away
+    /// @param previousOwner The previous owner of the OPWorldID contract
+    /// @param newOwner The new owner of the OPWorldID contract
+    /// @param isLocal Whether the ownership transfer is local (Optimism/OP Stack chain EOA/contract)
+    /// or an Ethereum EOA or contract
+    event OwnershipTransferredOp(
+        address indexed previousOwner, address indexed newOwner, bool isLocal
+    );
+
+    /// @notice Emmitted when the the StateBridge sends a root to the OPWorldID contract
+    /// @param root The root sent to the OPWorldID contract on the OP Stack chain
+    event RootPropagated(uint256 root);
+
+    /// @notice Emmitted when the the StateBridge sets the root history expiry for OpWorldID and PolygonWorldID
+    /// @param rootHistoryExpiry The new root history expiry
+    event SetRootHistoryExpiryOp(uint256 rootHistoryExpiry);
+
+    /// @notice Emmitted when the the StateBridge sets the gas limit for sendRootOp
+    /// @param _opGasLimit The new opGasLimit for sendRootOp
+    event SetGasLimitSendRootOp(uint32 _opGasLimit);
+
+    /// @notice Emmitted when the the StateBridge sets the gas limit for setRootHistoryExpiryOpt
+    /// @param _opGasLimit The new opGasLimit for setRootHistoryExpiryOptimism
+    event SetGasLimitSetRootHistoryExpiryOp(uint32 _opGasLimit);
+
+    /// @notice Emmitted when the the StateBridge sets the gas limit for transferOwnershipOp
+    /// @param _opGasLimit The new opGasLimit for transferOwnershipOptimism
+    event SetGasLimitTransferOwnershipOp(uint32 _opGasLimit);
```

Comment:
Just a thought, when deploying to other Op stack chains like Base, will we be changing the `Op` suffix to `Base`? If we are planning on doing any monitoring for these events, it might be helpful to remove the suffix and keep the event signature the same for ease of indexing.

Reply:
ok, let's keep the fn signature the same, the reason for keeping op was because Op as in OP stack, but doesn't really matter I'll remove

Reply:
I only decided to keep `transferOwnershipOp` so as to not conflict with `transferOwnership` from `Ownable2Step`.